### PR TITLE
kdebug_signpost* fns don't exist on macOS 10.11

### PR DESCRIPTION
**Pull request by datadog23**
> Fixes #issue
Doesn't fix, but found during https://github.com/gfx-rs/gfx/issues/2563

> PR checklist:
> - [ ] `make` succeeds (on *nix)
Yes.

> - [ ] `make reftests` succeeds
No, as it seems to be trying to compile with the feature this patch adds ('signpost') turned on?

> - [ ] tested examples with the following backends:
metal

> - [ ] `rustfmt` run on changed code
I don't know how, but I tried to copy the existing style.

### CODE REVIEW

File: src/warden/Cargo.toml [hunk 1]
```diff
@@ -51,6 +51,12 @@ version = "0.2"
 features = ["auto-capture"]
 optional = true
 
+[target.'cfg(target_os = "macos")'.dependencies.gfx-backend-metal]
+path = "../../src/backend/metal"
+version = "0.2"
+features = ["signpost"]
```
Comment (kvark) [inline src/warden/Cargo.toml]: I don't think Warden needs signposts

Comment (datadog23) [reply src/warden/Cargo.toml]: No, I was just blindly copying the "auto-capture" example, back when I couldn't work out how to pass feature flags to cargo.

**Comment by datadog23** [comment]
Um, do I need to do anything else here? I agree that changes to `warden/Cargo.toml` are unnecessary, do I need to submit another commit undoing those?

**Comment by kvark** [comment]
@datadog23 the feature code is good, but we are missing the CI coverage for it. So here are the small steps left here:
  1. remove the changes to Warden
  2. add the following line to the end of `.travis.yml`
  3. squash the commits together and update this PR

```yml
if [[ $TRAVIS_OS_NAME == "osx" ]]; then cd src/backend/metal && cargo check --all-features; fi
```

**Comment by kvark** [comment]
ping!

**Comment by kvark** [comment]
@datadog23 is anything unclear about the path forward?

**Comment by kvark** [comment]
@datadog23 without having a word from you for 3 weeks, I decided to step in and finish the PR. Sorry if that causes any inconvenience! Please feel free to visit our chat and talk about other possible contributions.
bors r+

**Comment by kvark** [comment]
Enabling all the features on CI reveals that we haven't tested some of them. This needs a bit more work.
bors r-

**Comment by kvark** [comment]
Had to update the dispatch code quite a bit...
bors r=compiler

**Comment by datadog23** [comment]
Sorry, i panicked and fled, didn't dare come on github or the discord channel for a month, maybe don't ask utter beginners to submit Pull Requests when they just want to ask why things aren't working / if they could ever work?

**Comment by kvark** [comment]
Thanks for coming back! I'm surprised you panicked though. Did we treat you incorrectly? I thought we were responsive and giving you clear directions on how to finish this.

**Comment by counterfactual-regret-minimization** [comment]
I freaked out about being asked to modify/repair the CI system, which i'd never heard of before: i thought the implication was that i'd broken it, when i only wanted to corroborate a bug report!

Sorry for causing you all extra work

**Pull request merged**